web3-functions/index.ts

```diff
@@ -0,0 +1,293 @@
+    debugLog(getDashLine());
+    printSignedDataPackage(dataPackageIndex, signedDataPackage);
+
+    let dataFeed = dataFeedIds.get(
+      signedDataPackage.dataPackage.dataPoints[0].dataFeedId
+    );
+
+    if (
+      dataFeed != undefined &&
+      dataFeed.symbol === signedDataPackage.dataPackage.dataPoints[0].dataFeedId
+    ) {
+      if (dataFeed.livePrice.eq(zero)) {
+        dataFeed.livePrice = BigNumber.from(
+          signedDataPackage.dataPackage.dataPoints[0].value
+        );
+      }
+    }
+    debugLog("Data feed: ", dataFeed);
+    dataPackageIndex++;
+  }
+
+  // Check if all data feeds are present
+  for (const dataFeed of dataFeedIds.values()) {
+    if (dataFeed.livePrice.eq(zero)) {
+      console.log("Data feed not found: ", dataFeed);
+      return {
+        canExec: false,
+        message: `Data feed not found: ${dataFeed.symbol}`,
+      };
+    }
+  }
+  debugLog(getDashLine());
+
+  // Retrieve stored prices and timestamps from the blockchain
+  for (const dataFeed of dataFeedIds.values()) {
+    [dataFeed.storedTimestamp, , dataFeed.storedPrice] = await wrappedOracle
+      .getLastUpdateDetails(dataFeed.id)
+      .catch(() => [BigNumber.from(0), 0, 0]);
+  }
+  // And print them out
+  debugLog("Stored prices and timestamps:");
+  printPrices(dataFeedIds);
+  console.log(getDashLine());
+
+  // Check price deviation and create an array for price feeds which needs to be updated
+  let priceFeedIdsToUpdate: string[] = [];
+  console.log("Price deviations and time elapsed since last update:");
+  console.log(getDashLine());
+  for (const dataFeed of dataFeedIds.values()) {
+    const priceDeviation = computePriceDeviation(
+      dataFeed.livePrice,
+      dataFeed.storedPrice,
+      DECIMALS
+    );
+    console.log(
+      `Price deviation for ${dataFeed.symbol}: ${priceDeviation.toString()}`
+    );
+    const deviationPrct = (priceDeviation.toNumber() / 10 ** DECIMALS) * 100;
+    console.log(`Deviation in %: ${deviationPrct.toFixed(2)}%`);
+    debugLog(getDashLine());
+
+    // Check update time interval
+    const currentTimestamp = Date.now();
+    const timeElapsed =
+      (currentTimestamp - dataFeed.storedTimestamp) / (1000 * 60 * 60);
+    printTimestamps(dataFeed, currentTimestamp, timeElapsed);
+    console.log(getDashLine());
+
+    // Only update price if deviation is above 0.5% or last update is more than 6 hours ago
+    if (deviationPrct >= MIN_DEVIATION || timeElapsed >= MIN_TIME_ELAPSED) {
+      priceFeedIdsToUpdate.push(dataFeed.id);
+    }
+  }
+
+  // Print out the price feeds which needs to be updated as symbols
+  console.log("Price feeds to update: ", priceFeedIdsToUpdate);
+
+  // Only update price if deviation is above 0.5% or last update is more than 6 hours ago
+  if (priceFeedIdsToUpdate.length === 0) {
+    return {
+      canExec: false,
+      message: `No update: price deviation too small or time elapsed since last update is less than 6 hours`,
```

Comment:
```suggestion
      message: `No update: price deviation less than ${MIN_DEVIATION.toFixed(2)}% or time elapsed since last update is less than ${MAX_TIME_ELAPSED_HOURS} hours`,
```